modules/data-entry/src/main/frontend/src/dataHomepage/NewFormDialog.jsx

```diff
@@ -0,0 +1,116 @@
+    setError("");
+    
+    // Make a POST request to create a new form, with a randomly generated UUID
+    const URL = "/Forms/" + uuid();
+    var request_data = new FormData();
+    request_data.append('jcr:primaryType', 'lfs:Form');
+    request_data.append('questionnaire', questionnaireReference);
+    request_data.append('questionnaire@TypeHint', 'Reference');
+    fetch( URL, { method: 'POST', body: request_data })
+      .then( (response) => {
+        if (response.ok) {
+          // Redirect the user to the new uuid
+          // FIXME: Would be better to somehow obtain the router prefix from props
+          // but that is not currently possible
+          props.history.push("/content.html" + URL);
+        } else {
+          return(Promise.reject(response));
+        }
+      })
+      .catch(parseErrorResponse)
+      .finally(setFetching(false));
+    setFetching(true);
+  }
+
+  let openDialog = () => {
+    // Determine what questionnaires are available
+    setOpen(true);
+    setError("");
+    if (questionnaires.length === 0) {
+      // Send a fetch request to determine the questionnaires available
+      fetch('/query?query=' + encodeURIComponent('select * from [lfs:Questionnaire]'))
+        .then((response) => response.ok ? response.json() : Promise.reject(response))
+        .then((json) => {setQuestionnaires(json)})
+        .catch(parseErrorResponse)
+        .finally(setFetching(false));
+      setFetching(true);
+    }
+  }
+
+  // Parse an errored response object
+  let parseErrorResponse = (response) => {
+    setError(`New form request failed with error code ${response.status}: ${response.statusText}`);
+  }
+
+  return (
+    <React.Fragment>
+      { /* Only create a dialog if we need to allow the user to choose from multiple questionnaires */
+      (!presetPath) && <Dialog open={open} onClose={() => { setOpen(false); }}>
+        <DialogTitle id="new-form-title">
+          Select questionnaire
+        </DialogTitle>
+        {error && <Typography color='error'>{error}</Typography>}
+        {questionnaires &&
+          <List>
+            {questionnaires.map((questionnaire) => {
+              return (
+              <ListItem button key={questionnaire["jcr:uuid"]} onClick={() => {createForm(questionnaire["@path"])}} disabled={isFetching}>
+                <ListItemText primary={questionnaire["title"]}>
+                </ListItemText>
+              </ListItem>);
+            })}
+          </List>
+        }
+        {isFetching && <CircularProgress size={24} className={classes.newFormCircularProgress} />}
+      </Dialog>}
+      <div className={classes.newFormButtonWrapper}>
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={presetPath ? () => {createForm(presetPath)} : openDialog}
+          disabled={isFetching}
+          >
+          { children }
+        </Button>
+        {<CircularProgress size={24} className={classes.newFormCircularProgress} />}
```

Comment:
Shouldn't the loading indicator be shown only while waiting for a response?
```suggestion
        {isFetching && <CircularProgress size={24} className={classes.newFormLoadingIndicator} />}
```